### DISCUSSION TITLE
Slider: Is there a way to change the content of the aria-valuetext property?

### DISCUSSION THREAD

**Discussion by AlmeroSteyn** [adobe/react-spectrum#6316] [Q&A] [answered]
I would like to have all screen readers read out the value of the Slider as a percentage.

Is there a way to manipulate the aria-valuetext property written on the range input?

**Comment by reidbarber** [comment]
You should be able to pass in

```jsx
formatOptions={{style: 'percent'}}
```

and have the percent included in the aria-valuetext.

**Reply by AlmeroSteyn** [reply]
Thanks for taking time to respond! I appreciate that!

It does add % to aria-valuetext. Somehow I missed this in the docs, sorry.

BUT it formats the percentage as thousands. So when it is at 50%, the value text says 5000% for some reason. Min is set as 0 and max as 100 on the range input in the HTML so this is kinda surprising. Will need to see if I can figure out why later.

**Reply by reidbarber** [reply]
Interesting, feel free to create a codesandbox/stackblitz if you want us to take a closer look at that example.

**Reply by AlmeroSteyn** [reply]
It is already reproducible in the Storybook example in the docs as it turns out:

https://react-spectrum.adobe.com/react-aria-starter/index.html?path=/docs/slider--docs

If I set the formatOptions prop to:

```
{
  "style": "percent"
}
```

It changes the aria-valuetext to thousands of percent:

<img width="372" alt="Image showing slider with output formatted to thousands of percent" src="https://github.com/adobe/react-spectrum/assets/5063473/9598a020-08fa-49be-8032-3fe9f2a49359">

For that matter, it also formats the Output to that. I reformatted the output in my implementation so I did not notice it before this.

I confirmed that my browser is set to US English so the thousands separator is comma.

**Reply by reidbarber** [reply, marked as answer]
Oh, I think I see what is happening. The percent format option is just formatting the value, it doesn't have any other context. So .5 = 50%,  1 = 100%, etc. This is just how Intl.NumberFormat() works, which formatOptions goes to.

I think you'll want to set minValue to 0, maxValue to 1, and bring down the step to something like 0.01.

<img width="485" alt="Screenshot 2024-05-07 at 9 52 53 AM" src="https://github.com/adobe/react-spectrum/assets/8961049/af25f65d-64e3-42f4-ab97-970bb174e2d8">

**Reply by AlmeroSteyn** [reply]
Yeah that kinda makes sense! 

Tried it locally and works. Does mean the value for the change event will need to be corrected to make the full solution work, but it will certainly give me what I want.

Thanks for the suggestion!
